core/extension_set.rs

```diff
@@ -274,29 +272,41 @@ pub fn into_sources(
         .push(FastString::from_static(esm_entry_point));
     }
     for file in &*extension.lazy_loaded_esm_files {
-      let code = load(transpiler, file, source_mapper, &mut load_callback)?;
+      let (code, maybe_source_map) =
+        load(transpiler, file, &mut load_callback)?;
       sources.lazy_esm.push(LoadedSource {
         source_type: ExtensionSourceType::LazyEsm,
         specifier: ModuleName::from_static(file.specifier),
         code,
       });
+      if let Some(source_map) = maybe_source_map {
+        source_maps.push((ModuleName::from_static(file.specifier), source_map));
+      }
```

Comment:
nit: maybe this can be a field on `LoadedSource` instead? slightly cleaner and saves a vec alloc. 
```suggestion
      sources.lazy_esm.push(LoadedSource {
        source_type: ExtensionSourceType::LazyEsm,
        specifier: ModuleName::from_static(file.specifier),
        source_map: maybe_source_map
        code,
      });
```